plugin/kubernetes/kubernetes.go

```diff
@@ -102,10 +102,10 @@ func (k *Kubernetes) Services(ctx context.Context, state request.Request, exact
 
 		segs := dns.SplitDomainName(t)
 		if len(segs) != 1 {
-			return nil, nil
+			return nil, errInvalidRequest
```

Comment:
The answer should be the same as in the next `if`.
This and the following `if` can be combined into one.

Reply:
I ended up restructuring this a bit, rather than checking for a negative (make sure this isn't a query for "dns-version"), we now check if it is a query for "dns-version" and serve that if so, otherwise return NXDOMAIN